File: front-2/src/app/domain/posts/Post.entity.ts
import { CommentEntity } from "../comments/Comment.entity";
import { TopicEntity } from "../topics/Topic.entity";
import { UserEntity } from "../user/User.entity";

export class PostEntity {
    id?: number;
    title: string;
    content: string;
    author: UserEntity;
    topic : TopicEntity;
    comments: CommentEntity[];
    createdAt?: Date;

    constructor(title: string, content: string, topic: TopicEntity, author: UserEntity , comments: CommentEntity[]) {
        this.title = title;
        this.content = content;
        this.topic = topic;
        this.author = author;
        this.comments = comments;
    }
}

File: back-api/src/main/java/com/openclassrooms/mddapi/models/PostEntity.java
package com.openclassrooms.mddapi.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "posts")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PostEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "title")
    private String title;
    @Column(name = "content")
    private String content;

    @ManyToOne
    @JoinColumn(name = "user_id")
    @JsonIgnoreProperties(value = "subscriptions")
    private UserEntity author;

    @ManyToOne
    @JoinColumn(name = "topic_id")
    private TopicEntity topic;

    @OneToMany(mappedBy = "post")
    private List<CommentEntity> comments;

    @Column(name = "created_at")
    private LocalDate createdAt;

}
